src/components/VolunteerAutomation/AttendClass.js

```diff
@@ -0,0 +1,333 @@
+  };
+
+  useEffect(() => {
+    dispatch(classActions.getClasses());
+  }, [dispatch]);
+
+  const languageMap = {
+    hi: "Hindi",
+    te: "Telugu",
+    en: "English",
+    ta: "Tamil",
+    cohort: "Batch",
+  };
+
+  const classData = JSON.parse(localStorage.getItem("classes"));
+
+  let sliceData = classData || [];
+  chooseClassAgain &&
+    data &&
+    data.slice(0, 6).map((item) => {
+      sliceData.push(item);
+    });
+
+  !chooseClassAgain &&
+    sliceData &&
+    sliceData.find((item) => {
+      if (item.id == enrollId) {
+        sliceData = [];
+        sliceData.push(item);
+        localStorage.setItem("classes", JSON.stringify(sliceData));
+      }
+    });
+
+  const enrollClass = (Class) => {
+    setOpen(false);
+    setEnrollId(Class.id);
+    localStorage.setItem("classes", JSON.stringify([Class]));
+    setChooseClassAgain(false);
+    const notify = () => {
+      toast.success("You have been enrolled to class successfully", {
+        position: toast.POSITION.BOTTOM_RIGHT,
+        autoClose: 2500,
+      });
+    };
+    axios
+      .post(
+        `${process.env.REACT_APP_MERAKI_URL}/classes/${Class.id}/register`,
+        {},
+        {
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: user.data.token,
+            "register-to-all": "false",
+          },
+        }
+      )
+      .then((res) => {
+        notify();
+      });
+  };
+
+  const dropOutClass = (Class) => {
+    setOpen(false);
+    setProceed(false);
+    localStorage.setItem("proceed", false);
+    setChooseClassAgain(true);
+    const notify = () => {
+      toast.success("You have been dropout the class successfully", {
+        position: toast.POSITION.BOTTOM_RIGHT,
+        autoClose: 2500,
+      });
+    };
+    return axios({
+      method: METHODS.DELETE,
+      url: `${process.env.REACT_APP_MERAKI_URL}/classes/${Class.id}/unregister`,
+      headers: {
+        accept: "application/json",
+        Authorization: user.data.token,
+        "unregister-all": "false",
+      },
+    }).then(() => {
+      sliceData = [];
+      data &&
+        data.slice(0, 6).map((item) => {
+          sliceData.push(item);
+        });
+      localStorage.setItem("classes", JSON.stringify(sliceData));
+      notify();
+    });
+  };
+
+  const EnrolledAndTimer = ({ item }) => {
+    const [Timer, setTimer] = useState(TimeLeft(item.start_time));
+    const ONE_MINUTE = 60000; //millisecs
+    setInterval(() => {
+      setTimer(TimeLeft(item.start_time));
+    }, ONE_MINUTE);
+    return (
+      <>
+        {Timer == "joinNow" ? (
+          <a
+            style={{
+              textDecoration: "none",
+            }}
+            href={item.meet_link}
+            target="_blank"
+          >
+            <Button
+              onClick={() => {
+                setProceed(true);
+                localStorage.setItem("proceed", true);
+                localStorage.setItem("disabled", false);
```

Comment:
Instead of keeping track of `proceed` and `disabled` separately, why don't we just check if they're enrolled in a class?